core/src/main/java/io/parsingdata/metal/data/Environment.java

```diff
@@ -0,0 +1,63 @@
+/*
+ * Copyright 2013-2016 Netherlands Forensic Institute
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.parsingdata.metal.data;
+
+import static io.parsingdata.metal.Util.checkNotNull;
+import static io.parsingdata.metal.token.Token.NO_NAME;
+
+import io.parsingdata.metal.data.callback.Callbacks;
+import io.parsingdata.metal.encoding.Encoding;
+
+public class Environment {
+
+    public final String scope;
+    public final ParseState parseState;
+    public final Callbacks callbacks;
+    public final Encoding encoding;
+
+    public Environment(final String scope, final ParseState parseState, final Callbacks callbacks, final Encoding encoding) {
```

Comment:
Just a thought: I think these constructors can be package private, but then the EnvironmentFactory must be moved to a different package (io.parsingdata.metal.data). I do think that a public Environment is undesired. At least there is/should be no need for users to create there own environment. Or am I missing an  possible scenario for which environment should be public?

Reply:
As discussed, the `EnvironmentFactory` is current a test class. So I won't make this change now -- but all the same it's a good discussion to perhaps make one as part of the production classes.